OpenLIFULib/OpenLIFULib/algorithm_input_widget.py

```diff
@@ -149,23 +154,47 @@ def update(self):
-        else:
-            self.target_input.combo_box.setEnabled(True)
-            for target_node in target_nodes:
-                self.target_input.combo_box.addItem(target_node.GetName(), target_node)
+        # Update target combo box if part of the algorithm inputs
+        if "Target" in self.inputs_dict.keys():
+            target_nodes = get_target_candidates()
+            if len(target_nodes) == 0:
+                self.inputs_dict["Target"].indicate_no_options()
+            else:
+                self.inputs_dict["Target"].combo_box.setEnabled(True)
+                for target_node in target_nodes:
+                    self.inputs_dict["Target"].combo_box.addItem(target_node.GetName(), target_node)
+
+        # Update photoscans combobox 
+        # NOTE: This code can be moved to populate_from_loaded_objects once photoscans are associated with sessions
+        # This is temporarily here to populate the combobox with the photoscan loaded to the scene in the
+        # transducer tracking module. This may change based on how openlifu-python handles photoscans. 
+        if "Photoscan" in self.inputs_dict.keys():
+            num_photoscans = 0
+            for model_node in slicer.util.getNodesByClass('vtkMRMLModelNode'):
+                # Check that the model is a loaded photoscan model
+                if model_node.GetAttribute('isOpenLIFUPhotoscan'):
+                    self.inputs_dict["Photoscan"].combo_box.setEnabled(True)
+                    self.add_photoscan_to_combobox(model_node)
+                    num_photoscans += 1
+            if num_photoscans == 0:
+                self.inputs_dict["Photoscan"].indicate_no_options()
 
         # Set selections to the previous ones when they exist
         self._set_most_recent_selections()
 
     def has_valid_selections(self) -> bool:
         """Whether all options have been selected, so that get_current_data would return
         a complete set of data with no `None`s."""
-        return all(input.combo_box.currentData is not None for input in self.inputs)
-
-    def get_current_data(self) -> Tuple[SlicerOpenLIFUProtocol,SlicerOpenLIFUTransducer,vtkMRMLScalarVolumeNode,vtkMRMLMarkupsFiducialNode]:
-        """Get the current selections as a tuple"""
-        return tuple(input.combo_box.currentData for input in self.inputs)
+        return all(input.combo_box.currentData is not None for input in self.inputs_dict.values())
+
+    def get_current_data(self) -> Dict[str, Any]:
+        """Get the current selections as a Dictionary. Potential output data types are:
+            Protocol: SlicerOpenLIFUProtocol
+            Transducer: SlicerOpenLIFUTransducer
+            Volume: vtkMRMLScalarVolumeNode
+            Target: vtkMRMLMarkupsFiducialNode
+            Photoscan: vtkMRMLModelNode """
+        current_data_dict = {}
+        for input in self.inputs_dict.values():
+            current_data_dict[input.name] = input.combo_box.currentData
```

Comment:
```suggestion
        current_data_dict = {
            input.name : input.combo_box.currentData
            for input in self.inputs_dict.values()
        }
```

(just style preference, no need to change)